src/pgduckdb_hooks.cpp

```diff
@@ -207,8 +203,21 @@ DuckdbPlannerHook_Cpp(Query *parse, const char *query_string, int cursor_options
 			}
 			/* If we can't create a plan, we'll fall back to Postgres */
 		}
+		if (parse->commandType != CMD_SELECT && pgduckdb::DuckdbDidWrites() && pgduckdb::IsInTransactionBlock(true)) {
```

Comment:
Reading statements like this, I feel it might be useful to either nest PG APIs in a sub-namespace `pg` or to include PG in their name like `PGIsInTransactionBlock`.
Since we're bridging two databases, it is helpful to be extra-pedentic when dealing with their APIs.

What do you think?

Reply:
Agreed, I was thinking the same. Done now, let me know what you think. We should probably update the other code in another PR for consistency.

Reply:
This looks really good, much clearer. And agreed to update the rest of the repo to follow the same philosophy.